apps/andi/lib/andi_web/plugs/redirect.ex

```diff
@@ -0,0 +1,15 @@
+defmodule AndiWeb.Redirect do
+  @moduledoc """
+  A Plug to allow for easily doing redirects within a Plug or Phoenix router.
+
+  Based on code found at:
+    https://www.viget.com/articles/how-to-redirect-from-the-phoenix-router/
+  """
+  def init([to: _] = opts), do: opts
+  def init(_default), do: raise("Missing required option ':to' in redirect")
```

Comment:
```elixir
  def init(opts), do: Keyword.fetch!(opts, :to)
```
While this won't give you the custom error message, this is the more typical way to ensure a keyword argument is passed.

Reply:
Okay - we'll break this into a function that checks for the existence of the key.
```
if Keyword.has_key?(opts) do
  opts
else
 raise "error"
end
```